features/call/impl/src/main/kotlin/io/element/android/features/call/impl/utils/ActiveCallManager.kt

```diff
@@ -281,73 +287,75 @@ class DefaultActiveCallManager(
 
     @OptIn(ExperimentalCoroutinesApi::class)
     private fun observeRingingCall() {
-        activeCall
-            .filterNotNull()
-            .filter { it.callState is CallState.Ringing && it.callType is CallType.RoomCall }
-            .flatMapLatest { activeCall ->
-                val callType = activeCall.callType as CallType.RoomCall
-                val ringingInfo = activeCall.callState as CallState.Ringing
-                val client = matrixClientProvider.getOrRestore(callType.sessionId).getOrNull() ?: run {
-                    Timber.tag(tag).d("Couldn't find session for incoming call: $activeCall")
-                    return@flatMapLatest flowOf()
-                }
-                val room = client.getRoom(callType.roomId) ?: run {
-                    Timber.tag(tag).d("Couldn't find room for incoming call: $activeCall")
-                    return@flatMapLatest flowOf()
-                }
+        val roomForActiveCallFlow: Flow<Pair<BaseRoom, EventId>?> = activeCall.mapLatest { activeCall ->
+            val callType = activeCall?.callType as? CallType.RoomCall ?: return@mapLatest null
+            val ringingInfo = activeCall.callState as? CallState.Ringing ?: return@mapLatest null
+            val client = matrixClientProvider.getOrRestore(callType.sessionId).getOrNull() ?: run {
+                Timber.tag(tag).d("Couldn't find session for incoming call: $activeCall")
+                return@mapLatest null
+            }
+            val room = client.getRoom(callType.roomId) ?: run {
```

Comment:
Room is never closed?

Reply:
With JNA we don't really need to, but I can make sure we close it manually.

Reply:
Or maybe I can't, since it's a value that's shared and doesn't have a clear lifetime 🤔 .